optimum/neuron/distributed/parallel_layers.py

```diff
@@ -705,3 +707,202 @@ def transform(
         cls.patch_cross_entropy(model)
 
         return layer
+
+
+@dataclass(frozen=True)
+class SequenceCollectiveOpInfo:
+    """
+    Represents a collective op to perform.
+
+    Example:
+
+        ```python
+        info = SequenceCollectiveOpInfo("gather", torch.nn.LayerNorm, "output", "last")
+        ```
+        Here `info` indicates that the output of the last `torch.nn.LayerNorm` must be gathered on the sequence axis.
+
+    """
```

Comment:
You should provide a description of the Args matching the type hints below, detailing the allowed values for collective_op, io.